File: src/oop/FileSearch/Search/Attributes/BySizeMatchCriteria.cs
using Microsoft.Extensions.FileProviders;

namespace FileSearch.Search.Attributes;

public sealed class BySizeMatchCriteria(long expectedValue, ComparisonOperandType comparisonOperandType)
    : ComparisonMatchCriteriaBase<long>(expectedValue, comparisonOperandType)
{
    protected override long GetAttributeValue(IFileInfo fileInfo, SearchOptions options)
        => fileInfo.Length;
}

File: src/python/oop/FileSearch/search/criterias/by_size_match_criteria.py
from .match_criteria_base import *
from ..file_info import *
from ..search_options import *

class BySizeMatchCriteria(MatchCriteriaBase[int]):
    def __init__(self, expectedSize: int):
        self.__expectedSize = expectedSize

    def is_match(self, file: FileInfo, options: SearchOptions = None) -> bool:
        return self.__expectedSize == self.__get_attribute(file)

    def __get_attribute(self, file: FileInfo) -> T:
        return file.size
